### PR TITLE
RI-7632: rework statistics for api/v2 client

### PR DESCRIPTION
**Pull request by ArtemHoruzhenko**
# What

This PR implements the RDI statistics functionality for the API v2 client, adding support for fetching and transforming statistics data from RDI API v2 endpoints.

### Backend (API)

- **New Statistics Client Methods**:
  - Added `getStatistics()` method to `ApiV2RdiClient` for fetching statistics from RDI API v2
  - Implemented proper error handling and response transformation
  - Integrated with existing RDI client architecture

- **Statistics Transformers**:
  - Created `RdiStatisticsTransformer` to convert API v2 responses to internal DTOs
  - Implemented transformation logic for:
    - Client connection statistics
    - Stream processing metrics
    - Performance data
    - Ingestion statistics
  - Added proper type mapping and data validation

- **DTOs and Interfaces**:
  - Defined API v2 statistics response interfaces
  - Created transformation mappings between API v2 format and internal format
  - Ensured type safety across the statistics pipeline

### Testing

- **Unit Tests**:
  - Added comprehensive tests for `ApiV2RdiClient.getStatistics()`
  - Created tests for `RdiStatisticsTransformer`
  - Covered edge cases and error scenarios
  - Verified proper data transformation

## Technical Decisions

1. **Transformer Pattern**: Used a dedicated transformer class to separate API response parsing from business logic, making the code more maintainable and testable

2. **Type Safety**: Defined strict TypeScript interfaces for API v2 responses to catch type mismatches early

3. **Error Handling**: Implemented robust error handling to gracefully handle API failures and malformed responses

4. **Separation of Concerns**: Kept the client focused on HTTP communication while delegating transformation logic to specialized transformers

## Dependencies

This PR builds on top of:
- PR #5353 (RDI statistics API rework) - provides the DTOs and models used by the transformer

## Integration

The statistics client integrates with:
- `RdiClientProvider` - uses this client to fetch statistics
- RDI statistics DTOs - transforms API responses into these models
- RDI statistics controller - serves transformed data to the frontend

# Testing

## Automation Tests
- [ ] Unit tests for `ApiV2RdiClient.getStatistics()` method
- [ ] Unit tests for `RdiStatisticsTransformer`
- [ ] Tests cover successful responses, error cases, and edge cases
- [ ] Integration tests with actual RDI API v2 endpoint

## Manual Testing
- [ ] Verify statistics are correctly fetched from RDI API v2
- [ ] Test with different RDI configurations
- [ ] Verify error handling when RDI is unavailable or returns errors
- [ ] Check data transformation accuracy

## Test Coverage
- New files have comprehensive unit test coverage
- All transformation logic is tested
- Error scenarios are covered

---

> [!NOTE]
> **Medium Risk**
> Adds a new v2 statistics fetch path and transformation logic, which could affect monitoring/UX if the v2 response shape differs or pipelines aren’t selected as expected. Also introduces typed handling of connection details (including `password` in the response interface), so reviewers should ensure no sensitive data is surfaced downstream.
> 
> **Overview**
> Adds API v2 statistics support by introducing `ApiV2RdiClient.getStatistics()`, which fetches `api/v2/pipelines/{name}/metric-collections` and returns an `RdiStatisticsResult` with *Success/Fail* status handling.
> 
> Introduces v2 response typing (`GetMetricsCollectionResponse`) and a v2 transformer that reuses v1 statistics transformers while extending *processing performance* with `transform_time_avg` and `write_time_avg`, with accompanying unit tests to validate section ordering/types and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 33b3db64ea0e7d005e323e56a879a900d6f92fa7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+export enum ComponentMetricsCollections {
+  processorMetrics = 'processor-metrics',
```
Comment (valkirilov) [inline redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts]: nit: Very very minor, but just for consistency - in most places we use capitalized enums, so better to stick to it.

File: redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+export interface ComponentMetricsResponse {
+  name: ComponentMetricsCollections;
+  component: string;
+  metrics: object;
+}
+
+export interface ProcessorMetricsResponse extends ComponentMetricsResponse {
+  name: ComponentMetricsCollections.processorMetrics;
+  metrics: {
+    processing_performance: {
```
Comment (valkirilov) [inline redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts:10]: I didn't compare the interfaces between v1 and v2, I just wonder whether we have matching parts or they're complete independant from each other? 

Because, if we have matching sub-parts, maybe we don't have to redeclare them here again?

Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts:10]: the root is different when statistics are almost the same but I would like to see complete picture in a single file. and moreover we might have some changes (e.g. additional fields will be added without breaking changes) which might require to modify response interface

File: redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts (hunk 1 above)
Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/client/api/v2/responses/metrics-collections.response.ts]: had to remove it at all in the second commit. name is inconsistent so we can't rely on it.
